ccl/logictestccl: TestTenantLogic failed

[ccl/logictestccl.TestTenantLogic failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2870787&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=2870787&tab=artifacts#/) on [master @ 3e543621a81be7953bee980a6493beec952c693d](https://github.com/cockroachdb/cockroach/commits/3e543621a81be7953bee980a6493beec952c693d):


```
=== RUN   TestTenantLogic
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTenantLogic046977194
    test_log_scope.go:74: use -show-logs to present logs inline
=== CONT  TestTenantLogic
    logic.go:3333: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTenantLogic046977194
--- FAIL: TestTenantLogic (590.65s)
=== RUN   TestTenantLogic/3node-tenant/distsql_automatic_stats
    logic.go:2270: condition failed to evaluate within 45s: 
        ../../sql/logictest/testdata/logic_test/distsql_automatic_stats:66: SELECT DISTINCT ON (column_names) statistics_name, column_names, row_count, distinct_count, null_count
        FROM [SHOW STATISTICS FOR TABLE data] ORDER BY column_names ASC, created DESC
        expected:
            statistics_name  column_names  row_count  distinct_count  null_count
            __auto__         {a,b,c}       1000       1000            0
            __auto__         {a,b}         1000       100             0
            __auto__         {a}           1000       10              0
            __auto__         {b}           1000       10              0
            __auto__         {c}           1000       10              0
            __auto__         {d}           1000       2               730
            
        but found (query options: "colnames,rowsort,retry" -> ignore the following ordering of rows) :
            statistics_name  column_names  row_count  distinct_count  null_count
            __auto__         {a}           1000       10              0
            __auto__         {a,b}         1000       100             0
            __auto__         {a,b,c}       1000       1000            0
            __auto__         {b}           1000       10              0
            __auto__         {c}           1000       10              0
            __auto__         {d}           1000       1               1000
        goroutine 1348744 [running]:
        runtime/debug.Stack(0xc002f35278, 0x5a9a880, 0xc0360c6fe0)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0x9f
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x5c86ae0, 0xc01dc54c00, 0xc002f35278)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:36 +0x6d
        github.com/cockroachdb/cockroach/pkg/sql/logictest.(*logicTest).processSubtest(0xc00d874400, 0x0, 0x0, 0xc03ce1d8f0, 0x0, 0xc01ef5ca00, 0x3f, 0x4cf2e4b, 0xc, 0x3, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:2270 +0x2267
        github.com/cockroachdb/cockroach/pkg/sql/logictest.(*logicTest).processTestFile(0xc00d874400, 0xc01ef5ca00, 0x3f, 0x4cf2e4b, 0xc, 0x3, 0x0, 0x0, 0x0, 0x0, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:1815 +0x7b8
        github.com/cockroachdb/cockroach/pkg/sql/logictest.(*logicTest).runFile(0xc00d874400, 0xc01ef5ca00, 0x3f, 0x4cf2e4b, 0xc, 0x3, 0x0, 0x0, 0x0, 0x0, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:3107 +0xd8
        github.com/cockroachdb/cockroach/pkg/sql/logictest.RunLogicTestWithDefaultConfig.func1.1(0xc01dc54c00)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:3306 +0x2de
        testing.tRunner(0xc01dc54c00, 0xc01def5000)
        	/usr/local/go/src/testing/testing.go:1123 +0xef
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:1168 +0x2b3
--- done: ../../sql/logictest/testdata/logic_test/distsql_automatic_stats with config 3node-tenant: 10 tests, 0 failures
E210412 05:19:48.914206 1350906 sql/stats/automatic_stats.go:302  [-] 50  failed to refresh stats: node unavailable; try another peer
        --- FAIL: TestTenantLogic/3node-tenant/distsql_automatic_stats (46.63s)
=== RUN   TestTenantLogic/3node-tenant
    --- FAIL: TestTenantLogic/3node-tenant (590.55s)
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
make stressrace TESTS=TestTenantLogic PKG=./pkg/ccl/logictestccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
</p>
<p>Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #62550 ccl/logictestccl: TestTenantLogic failed [C-test-failure O-robot branch-release-21.1]</p>
</details>

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantLogic.*&sort=title+created&display=lastcommented+project)

<sub>[Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub>

ccl/logictestccl.TestTenantLogic [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2996058&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=2996058&tab=artifacts#/) on master @ [617e1ed8b755fcb488fd90c15c69eb68ba1dd64e](https://github.com/cockroachdb/cockroach/commits/617e1ed8b755fcb488fd90c15c69eb68ba1dd64e):


```
    logic.go:3358: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTenantLogic883484109
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTenantLogic496797367
    test_log_scope.go:135: -- test log scope end --
    test_log_scope.go:74: use -show-logs to present logs inline
=== CONT  TestTenantLogic
    logic.go:3358: -- test log scope end --
--- FAIL: TestTenantLogic (577.06s)
=== RUN   TestTenantLogic/3node-tenant/distsql_automatic_stats
    logic.go:2264: condition failed to evaluate within 45s: 
        ../../sql/logictest/testdata/logic_test/distsql_automatic_stats:66: SELECT DISTINCT ON (column_names) statistics_name, column_names, row_count, distinct_count, null_count
        FROM [SHOW STATISTICS FOR TABLE data] ORDER BY column_names ASC, created DESC
        expected:
            statistics_name  column_names  row_count  distinct_count  null_count
            __auto__         {a,b,c}       1000       1000            0
            __auto__         {a,b}         1000       100             0
            __auto__         {a}           1000       10              0
            __auto__         {b}           1000       10              0
            __auto__         {c}           1000       10              0
            __auto__         {d}           1000       2               730
            
        but found (query options: "colnames,rowsort,retry" -> ignore the following ordering of rows) :
            statistics_name  column_names  row_count  distinct_count  null_count
            __auto__         {a}           1000       10              0
            __auto__         {a,b}         1000       100             0
            __auto__         {a,b,c}       1000       1000            0
            __auto__         {b}           1000       10              0
            __auto__         {c}           1000       10              0
            __auto__         {d}           1000       1               1000
        goroutine 1142108 [running]:
        runtime/debug.Stack(0xc00e337270, 0x5be4980, 0xc00509cc60)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0x9f
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x5cb3a80, 0xc03477ea80, 0xc00e337270)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:35 +0x6d
        github.com/cockroachdb/cockroach/pkg/sql/logictest.(*logicTest).processSubtest(0xc005153600, 0x0, 0x0, 0xc0001a4600, 0x0, 0xc023416a40, 0x3f, 0x4dfd4f7, 0xc, 0x3, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:2264 +0x2267
        github.com/cockroachdb/cockroach/pkg/sql/logictest.(*logicTest).processTestFile(0xc005153600, 0xc023416a40, 0x3f, 0x4dfd4f7, 0xc, 0x3, 0x0, 0x0, 0x0, 0x0, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:1809 +0x7b8
        github.com/cockroachdb/cockroach/pkg/sql/logictest.(*logicTest).runFile(0xc005153600, 0xc023416a40, 0x3f, 0x4dfd4f7, 0xc, 0x3, 0x0, 0x0, 0x0, 0x0, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:3128 +0xd8
        github.com/cockroachdb/cockroach/pkg/sql/logictest.RunLogicTestWithDefaultConfig.func1.1(0xc03477ea80)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:3331 +0x2de
        testing.tRunner(0xc03477ea80, 0xc009daf780)
        	/usr/local/go/src/testing/testing.go:1123 +0xef
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:1168 +0x2b3
--- done: ../../sql/logictest/testdata/logic_test/distsql_automatic_stats with config 3node-tenant: 10 tests, 0 failures
        --- FAIL: TestTenantLogic/3node-tenant/distsql_automatic_stats (46.69s)
=== RUN   TestTenantLogic/3node-tenant
    --- FAIL: TestTenantLogic/3node-tenant (574.10s)
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
make stressrace TESTS=TestTenantLogic PKG=./pkg/ccl/logictestccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
</p>
<p>Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #62550 ccl/logictestccl: TestTenantLogic failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
/cc @cockroachdb/sql-queries nvanbenschoten
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantLogic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/logictestccl.TestTenantLogic [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3039293&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3039293&tab=artifacts#/) on master @ [8939515baf9fee3b605ace650629c24096fc38fd](https://github.com/cockroachdb/cockroach/commits/8939515baf9fee3b605ace650629c24096fc38fd):


```
    logic.go:3358: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTenantLogic890968064
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTenantLogic631698328
    test_log_scope.go:135: -- test log scope end --
    test_log_scope.go:74: use -show-logs to present logs inline
=== CONT  TestTenantLogic
    logic.go:3358: -- test log scope end --
--- FAIL: TestTenantLogic (797.59s)
=== RUN   TestTenantLogic/3node-tenant/distsql_automatic_stats
    logic.go:2264: condition failed to evaluate within 45s: 
        ../../sql/logictest/testdata/logic_test/distsql_automatic_stats:66: SELECT DISTINCT ON (column_names) statistics_name, column_names, row_count, distinct_count, null_count
        FROM [SHOW STATISTICS FOR TABLE data] ORDER BY column_names ASC, created DESC
        expected:
            statistics_name  column_names  row_count  distinct_count  null_count
            __auto__         {a,b,c}       1000       1000            0
            __auto__         {a,b}         1000       100             0
            __auto__         {a}           1000       10              0
            __auto__         {b}           1000       10              0
            __auto__         {c}           1000       10              0
            __auto__         {d}           1000       2               730
            
        but found (query options: "colnames,rowsort,retry" -> ignore the following ordering of rows) :
            statistics_name  column_names  row_count  distinct_count  null_count
            __auto__         {a}           1000       10              0
            __auto__         {a,b}         1000       100             0
            __auto__         {a,b,c}       1000       1000            0
            __auto__         {b}           1000       10              0
            __auto__         {c}           1000       10              0
            __auto__         {d}           1000       1               1000
        goroutine 1392073 [running]:
        runtime/debug.Stack(0xc010ea1270, 0x5d47820, 0xc027ae7740)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0x9f
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x5e192e0, 0xc0079eb080, 0xc010ea1270)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:35 +0x6d
        github.com/cockroachdb/cockroach/pkg/sql/logictest.(*logicTest).processSubtest(0xc00db2f400, 0x0, 0x0, 0xc0356e5140, 0x0, 0xc010af6a40, 0x3f, 0x4f3d3cc, 0xc, 0x3, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:2264 +0x2267
        github.com/cockroachdb/cockroach/pkg/sql/logictest.(*logicTest).processTestFile(0xc00db2f400, 0xc010af6a40, 0x3f, 0x4f3d3cc, 0xc, 0x3, 0x0, 0x0, 0x0, 0x0, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:1809 +0x7b8
        github.com/cockroachdb/cockroach/pkg/sql/logictest.(*logicTest).runFile(0xc00db2f400, 0xc010af6a40, 0x3f, 0x4f3d3cc, 0xc, 0x3, 0x0, 0x0, 0x0, 0x0, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:3128 +0xd8
        github.com/cockroachdb/cockroach/pkg/sql/logictest.RunLogicTestWithDefaultConfig.func1.1(0xc0079eb080)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:3331 +0x2de
        testing.tRunner(0xc0079eb080, 0xc00f54a180)
        	/usr/local/go/src/testing/testing.go:1123 +0xef
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:1168 +0x2b3
--- done: ../../sql/logictest/testdata/logic_test/distsql_automatic_stats with config 3node-tenant: 10 tests, 0 failures
        --- FAIL: TestTenantLogic/3node-tenant/distsql_automatic_stats (47.20s)
=== RUN   TestTenantLogic/3node-tenant
    --- FAIL: TestTenantLogic/3node-tenant (794.59s)
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
make stressrace TESTS=TestTenantLogic PKG=./pkg/ccl/logictestccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
</p>
<p>Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #62550 ccl/logictestccl: TestTenantLogic failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
/cc @cockroachdb/sql-queries nvanbenschoten
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantLogic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/logictestccl.TestTenantLogic [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3108421&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3108421&tab=artifacts#/) on master @ [b35fe177c0fb6bf58ba00357e152ecaf6c1e0a2a](https://github.com/cockroachdb/cockroach/commits/b35fe177c0fb6bf58ba00357e152ecaf6c1e0a2a):


```
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTenantLogic374557721
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTenantLogic795403665
    test_log_scope.go:135: -- test log scope end --
    test_log_scope.go:74: use -show-logs to present logs inline
=== CONT  TestTenantLogic
    logic.go:3355: -- test log scope end --
--- FAIL: TestTenantLogic (1404.37s)
=== RUN   TestTenantLogic/3node-tenant/distsql_automatic_stats
--- progress: ../../sql/logictest/testdata/logic_test/distsql_automatic_stats: 10 statements/queries
    logic.go:2264: condition failed to evaluate within 45s: 
        ../../sql/logictest/testdata/logic_test/distsql_automatic_stats:66: SELECT DISTINCT ON (column_names) statistics_name, column_names, row_count, distinct_count, null_count
        FROM [SHOW STATISTICS FOR TABLE data] ORDER BY column_names ASC, created DESC
        expected:
            statistics_name  column_names  row_count  distinct_count  null_count
            __auto__         {a,b,c}       1000       1000            0
            __auto__         {a,b}         1000       100             0
            __auto__         {a}           1000       10              0
            __auto__         {b}           1000       10              0
            __auto__         {c}           1000       10              0
            __auto__         {d}           1000       2               730
            
        but found (query options: "colnames,rowsort,retry" -> ignore the following ordering of rows) :
            statistics_name  column_names  row_count  distinct_count  null_count
            __auto__         {a}           1000       10              0
            __auto__         {a,b}         1000       100             0
            __auto__         {a,b,c}       1000       1000            0
            __auto__         {b}           1000       10              0
            __auto__         {c}           1000       10              0
            __auto__         {d}           1000       1               1000
        goroutine 1992582 [running]:
        runtime/debug.Stack(0xc00cff1278, 0x63da560, 0xc03ade52c0)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0x9f
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x6494e98, 0xc091e10300, 0xc00cff1278)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:35 +0x70
        github.com/cockroachdb/cockroach/pkg/sql/logictest.(*logicTest).processSubtest(0xc010566a00, 0x0, 0x0, 0xc037d59c50, 0x0, 0xc06053ca80, 0x3f, 0x4ed988c, 0xc, 0x3, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:2264 +0x2230
        github.com/cockroachdb/cockroach/pkg/sql/logictest.(*logicTest).processTestFile(0xc010566a00, 0xc06053ca80, 0x3f, 0x4ed988c, 0xc, 0x3, 0x0, 0x0, 0x0, 0x0, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:1809 +0x7b8
        github.com/cockroachdb/cockroach/pkg/sql/logictest.(*logicTest).runFile(0xc010566a00, 0xc06053ca80, 0x3f, 0x4ed988c, 0xc, 0x3, 0x0, 0x0, 0x0, 0x0, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:3128 +0xd8
        github.com/cockroachdb/cockroach/pkg/sql/logictest.RunLogicTestWithDefaultConfig.func1.1(0xc091e10300)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:3328 +0x2de
        testing.tRunner(0xc091e10300, 0xc006da2dc0)
        	/usr/local/go/src/testing/testing.go:1193 +0xef
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:1238 +0x2b3
--- done: ../../sql/logictest/testdata/logic_test/distsql_automatic_stats with config 3node-tenant: 10 tests, 0 failures
        --- FAIL: TestTenantLogic/3node-tenant/distsql_automatic_stats (49.26s)
=== RUN   TestTenantLogic/3node-tenant
    --- FAIL: TestTenantLogic/3node-tenant (1399.89s)
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
make stressrace TESTS=TestTenantLogic PKG=./pkg/ccl/logictestccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
</p>
<p>Parameters in this failure:

- GOFLAGS=-json
</p>
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #62550 ccl/logictestccl: TestTenantLogic failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
/cc @cockroachdb/sql-queries nvanbenschoten
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantLogic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/logictestccl.TestTenantLogic [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3250524&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3250524&tab=artifacts#/) on master @ [c995342ead51e08f8ed1155de4218d30a00d86d2](https://github.com/cockroachdb/cockroach/commits/c995342ead51e08f8ed1155de4218d30a00d86d2):


```
E210802 05:45:50.050221 1696646 server/server_sql.go:305  [-] 289 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:1627
E210802 05:45:50.050221 1696646 server/server_sql.go:305  [-] 289 +  | github.com/cockroachdb/cockroach/pkg/sql/logictest.RunLogicTestWithDefaultConfig.func1.1
E210802 05:45:50.050221 1696646 server/server_sql.go:305  [-] 289 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:3325
E210802 05:45:50.050221 1696646 server/server_sql.go:305  [-] 289 +  | testing.tRunner
E210802 05:45:50.050221 1696646 server/server_sql.go:305  [-] 289 +  | 	/usr/local/go/src/testing/testing.go:1193
E210802 05:45:50.050221 1696646 server/server_sql.go:305  [-] 289 +  | runtime.goexit
E210802 05:45:50.050221 1696646 server/server_sql.go:305  [-] 289 +  | 	/usr/local/go/src/runtime/asm_amd64.s:1371
E210802 05:45:50.050221 1696646 server/server_sql.go:305  [-] 289 +Wraps: (2) directory to store dumps could not be determined
E210802 05:45:50.050221 1696646 server/server_sql.go:305  [-] 289 +Error types: (1) *withstack.withStack (2) *errutil.leafError
    logic.go:2262: condition failed to evaluate within 45s: 
        ../../sql/logictest/testdata/logic_test/distsql_automatic_stats:66: SELECT DISTINCT ON (column_names) statistics_name, column_names, row_count, distinct_count, null_count
        FROM [SHOW STATISTICS FOR TABLE data] ORDER BY column_names ASC, created DESC
        expected:
            statistics_name  column_names  row_count  distinct_count  null_count
            __auto__         {a,b,c}       1000       1000            0
            __auto__         {a,b}         1000       100             0
            __auto__         {a}           1000       10              0
            __auto__         {b}           1000       10              0
            __auto__         {c}           1000       10              0
            __auto__         {d}           1000       2               730
            
        but found (query options: "colnames,rowsort,retry" -> ignore the following ordering of rows) :
            statistics_name  column_names  row_count  distinct_count  null_count
            __auto__         {a}           1000       10              0
            __auto__         {a,b}         1000       100             0
            __auto__         {a,b,c}       1000       1000            0
            __auto__         {b}           1000       10              0
            __auto__         {c}           1000       10              0
            __auto__         {d}           1000       1               1000
        goroutine 1696646 [running]:
        runtime/debug.Stack(0xc0106d3278, 0x6624020, 0xc0050488d0)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0x9f
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x66e22d8, 0xc0145dbc80, 0xc0106d3278)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:35 +0x70
        github.com/cockroachdb/cockroach/pkg/sql/logictest.(*logicTest).processSubtest(0xc01569e000, 0x0, 0x0, 0xc00472b2f0, 0x0, 0xc002376a00, 0x3f, 0x508a83a, 0xc, 0x3, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:2262 +0x2230
        github.com/cockroachdb/cockroach/pkg/sql/logictest.(*logicTest).processTestFile(0xc01569e000, 0xc002376a00, 0x3f, 0x508a83a, 0xc, 0x3, 0x0, 0x0, 0x0, 0x0, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:1807 +0x7b8
        github.com/cockroachdb/cockroach/pkg/sql/logictest.(*logicTest).runFile(0xc01569e000, 0xc002376a00, 0x3f, 0x508a83a, 0xc, 0x3, 0x0, 0x0, 0x0, 0x0, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:3126 +0xd8
        github.com/cockroachdb/cockroach/pkg/sql/logictest.RunLogicTestWithDefaultConfig.func1.1(0xc0145dbc80)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:3326 +0x2de
        testing.tRunner(0xc0145dbc80, 0xc014487080)
        	/usr/local/go/src/testing/testing.go:1193 +0xef
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:1238 +0x2b3
--- done: ../../sql/logictest/testdata/logic_test/distsql_automatic_stats with config 3node-tenant: 10 tests, 0 failures
        --- FAIL: TestTenantLogic/3node-tenant/distsql_automatic_stats (47.39s)
=== RUN   TestTenantLogic/3node-tenant
    --- FAIL: TestTenantLogic/3node-tenant (893.17s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestTenantLogic PKG=./pkg/ccl/logictestccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #62550 ccl/logictestccl: TestTenantLogic failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
/cc @cockroachdb/sql-queries nvanbenschoten
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantLogic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/logictestccl.TestTenantLogic [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3254635&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3254635&tab=artifacts#/) on master @ [1c46e1cd4e5be986bf9d13799bb7e13ddc896ed2](https://github.com/cockroachdb/cockroach/commits/1c46e1cd4e5be986bf9d13799bb7e13ddc896ed2):


```
E210803 05:44:53.427756 1619041 server/server_sql.go:306  [-] 291 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:1627
E210803 05:44:53.427756 1619041 server/server_sql.go:306  [-] 291 +  | github.com/cockroachdb/cockroach/pkg/sql/logictest.RunLogicTestWithDefaultConfig.func1.1
E210803 05:44:53.427756 1619041 server/server_sql.go:306  [-] 291 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:3325
E210803 05:44:53.427756 1619041 server/server_sql.go:306  [-] 291 +  | testing.tRunner
E210803 05:44:53.427756 1619041 server/server_sql.go:306  [-] 291 +  | 	/usr/local/go/src/testing/testing.go:1193
E210803 05:44:53.427756 1619041 server/server_sql.go:306  [-] 291 +  | runtime.goexit
E210803 05:44:53.427756 1619041 server/server_sql.go:306  [-] 291 +  | 	/usr/local/go/src/runtime/asm_amd64.s:1371
E210803 05:44:53.427756 1619041 server/server_sql.go:306  [-] 291 +Wraps: (2) directory to store dumps could not be determined
E210803 05:44:53.427756 1619041 server/server_sql.go:306  [-] 291 +Error types: (1) *withstack.withStack (2) *errutil.leafError
    logic.go:2262: condition failed to evaluate within 45s: 
        ../../sql/logictest/testdata/logic_test/distsql_automatic_stats:66: SELECT DISTINCT ON (column_names) statistics_name, column_names, row_count, distinct_count, null_count
        FROM [SHOW STATISTICS FOR TABLE data] ORDER BY column_names ASC, created DESC
        expected:
            statistics_name  column_names  row_count  distinct_count  null_count
            __auto__         {a,b,c}       1000       1000            0
            __auto__         {a,b}         1000       100             0
            __auto__         {a}           1000       10              0
            __auto__         {b}           1000       10              0
            __auto__         {c}           1000       10              0
            __auto__         {d}           1000       2               730
            
        but found (query options: "colnames,rowsort,retry" -> ignore the following ordering of rows) :
            statistics_name  column_names  row_count  distinct_count  null_count
            __auto__         {a}           1000       10              0
            __auto__         {a,b}         1000       100             0
            __auto__         {a,b,c}       1000       1000            0
            __auto__         {b}           1000       10              0
            __auto__         {c}           1000       10              0
            __auto__         {d}           1000       1               1000
        goroutine 1619041 [running]:
        runtime/debug.Stack(0xc00b98f278, 0x662eea0, 0xc02e57fd10)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0x9f
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x66ed258, 0xc00a3bcf00, 0xc00b98f278)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:35 +0x70
        github.com/cockroachdb/cockroach/pkg/sql/logictest.(*logicTest).processSubtest(0xc008faaa00, 0x0, 0x0, 0xc007d16420, 0x0, 0xc013312a00, 0x3f, 0x5093ca1, 0xc, 0x3, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:2262 +0x2230
        github.com/cockroachdb/cockroach/pkg/sql/logictest.(*logicTest).processTestFile(0xc008faaa00, 0xc013312a00, 0x3f, 0x5093ca1, 0xc, 0x3, 0x0, 0x0, 0x0, 0x0, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:1807 +0x7b8
        github.com/cockroachdb/cockroach/pkg/sql/logictest.(*logicTest).runFile(0xc008faaa00, 0xc013312a00, 0x3f, 0x5093ca1, 0xc, 0x3, 0x0, 0x0, 0x0, 0x0, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:3126 +0xd8
        github.com/cockroachdb/cockroach/pkg/sql/logictest.RunLogicTestWithDefaultConfig.func1.1(0xc00a3bcf00)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:3326 +0x2de
        testing.tRunner(0xc00a3bcf00, 0xc005bc8e40)
        	/usr/local/go/src/testing/testing.go:1193 +0xef
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:1238 +0x2b3
--- done: ../../sql/logictest/testdata/logic_test/distsql_automatic_stats with config 3node-tenant: 10 tests, 0 failures
        --- FAIL: TestTenantLogic/3node-tenant/distsql_automatic_stats (47.57s)
=== RUN   TestTenantLogic/3node-tenant
    --- FAIL: TestTenantLogic/3node-tenant (900.83s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestTenantLogic PKG=./pkg/ccl/logictestccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #62550 ccl/logictestccl: TestTenantLogic failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
/cc @cockroachdb/sql-queries nvanbenschoten
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantLogic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/logictestccl.TestTenantLogic [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3352057&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3352057&tab=artifacts#/) on master @ [967ed00f80981ce8848a5e8144ee6fbd29bc95bb](https://github.com/cockroachdb/cockroach/commits/967ed00f80981ce8848a5e8144ee6fbd29bc95bb):


```
    logic.go:3344: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTenantLogic610246766
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTenantLogic641059109
    test_log_scope.go:135: -- test log scope end --
    test_log_scope.go:74: use -show-logs to present logs inline
=== CONT  TestTenantLogic
    logic.go:3344: -- test log scope end --
--- FAIL: TestTenantLogic (1445.94s)
=== RUN   TestTenantLogic/3node-tenant/distsql_automatic_stats
    logic.go:2261: condition failed to evaluate within 45s: 
        ../../sql/logictest/testdata/logic_test/distsql_automatic_stats:66: SELECT DISTINCT ON (column_names) statistics_name, column_names, row_count, distinct_count, null_count
        FROM [SHOW STATISTICS FOR TABLE data] ORDER BY column_names ASC, created DESC
        expected:
            statistics_name  column_names  row_count  distinct_count  null_count
            __auto__         {a,b,c}       1000       1000            0
            __auto__         {a,b}         1000       100             0
            __auto__         {a}           1000       10              0
            __auto__         {b}           1000       10              0
            __auto__         {c}           1000       10              0
            __auto__         {d}           1000       2               730
            
        but found (query options: "colnames,rowsort,retry" -> ignore the following ordering of rows) :
            statistics_name  column_names  row_count  distinct_count  null_count
            __auto__         {a}           1000       10              0
            __auto__         {a,b}         1000       100             0
            __auto__         {a,b,c}       1000       1000            0
            __auto__         {b}           1000       10              0
            __auto__         {c}           1000       10              0
            __auto__         {d}           1000       1               1000
        goroutine 4755145 [running]:
        runtime/debug.Stack(0xc0099f9278, 0x6896700, 0xc05e071d70)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0x9f
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x6959458, 0xc06dcc0000, 0xc0099f9278)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:35 +0x70
        github.com/cockroachdb/cockroach/pkg/sql/logictest.(*logicTest).processSubtest(0xc0299fa600, 0x0, 0x0, 0xc019790810, 0x0, 0xc055dcd0c0, 0x3f, 0x527b72f, 0xc, 0x3, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:2261 +0x2230
        github.com/cockroachdb/cockroach/pkg/sql/logictest.(*logicTest).processTestFile(0xc0299fa600, 0xc055dcd0c0, 0x3f, 0x527b72f, 0xc, 0x3, 0x0, 0x0, 0x0, 0x0, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:1806 +0x7b8
        github.com/cockroachdb/cockroach/pkg/sql/logictest.(*logicTest).runFile(0xc0299fa600, 0xc055dcd0c0, 0x3f, 0x527b72f, 0xc, 0x3, 0x0, 0x0, 0x0, 0x0, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:3114 +0xd8
        github.com/cockroachdb/cockroach/pkg/sql/logictest.RunLogicTestWithDefaultConfig.func1.1(0xc06dcc0000)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:3317 +0x2de
        testing.tRunner(0xc06dcc0000, 0xc046198180)
        	/usr/local/go/src/testing/testing.go:1193 +0xef
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:1238 +0x2b3
--- done: ../../sql/logictest/testdata/logic_test/distsql_automatic_stats with config 3node-tenant: 10 tests, 0 failures
        --- FAIL: TestTenantLogic/3node-tenant/distsql_automatic_stats (48.98s)
=== RUN   TestTenantLogic/3node-tenant
    --- FAIL: TestTenantLogic/3node-tenant (1436.32s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestTenantLogic PKG=./pkg/ccl/logictestccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #62550 ccl/logictestccl: TestTenantLogic failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
/cc @cockroachdb/sql-queries nvanbenschoten
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantLogic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/logictestccl.TestTenantLogic [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3496835&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3496835&tab=artifacts#/) on master @ [9ba8499e80a3234da094e061827f1c23d9d33341](https://github.com/cockroachdb/cockroach/commits/9ba8499e80a3234da094e061827f1c23d9d33341):


```
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTenantLogic164665724
    test_log_scope.go:148: -- test log scope end --
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestTenantLogic
    logic.go:3436: -- test log scope end --
--- FAIL: TestTenantLogic (765.86s)
=== RUN   TestTenantLogic/3node-tenant/distsql_automatic_stats
    logic.go:2311: condition failed to evaluate within 45s: 
        ../../sql/logictest/testdata/logic_test/distsql_automatic_stats:66: SELECT DISTINCT ON (column_names) statistics_name, column_names, row_count, distinct_count, null_count
        FROM [SHOW STATISTICS FOR TABLE data] ORDER BY column_names ASC, created DESC
        expected:
            statistics_name  column_names  row_count  distinct_count  null_count
            __auto__         {a,b,c}       1000       1000            0
            __auto__         {a,b}         1000       100             0
            __auto__         {a}           1000       10              0
            __auto__         {b}           1000       10              0
            __auto__         {c}           1000       10              0
            __auto__         {d}           1000       2               730
            
        but found (query options: "colnames,rowsort,retry" -> ignore the following ordering of rows) :
            statistics_name  column_names  row_count  distinct_count  null_count
            __auto__         {a}           1000       10              0
            __auto__         {a,b}         1000       100             0
            __auto__         {a,b,c}       1000       1000            0
            __auto__         {b}           1000       10              0
            __auto__         {c}           1000       10              0
            __auto__         {d}           1000       1               1000
        goroutine 1397239 [running]:
        runtime/debug.Stack(0xc02cd21270, 0xa7a358200, 0x6a10fe0)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0x9f
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithin(0x6ad5ed8, 0xc01fac2c00, 0xc02cd21270, 0xa7a358200)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:60 +0x7b
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x6ad5ed8, 0xc01fac2c00, 0xc02cd21270)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:41 +0x65
        github.com/cockroachdb/cockroach/pkg/sql/logictest.(*logicTest).processSubtest(0xc0539d0200, 0x0, 0x0, 0xc00a969cb0, 0x0, 0xc0037169c0, 0x3f, 0x5319b22, 0xc, 0x3, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:2311 +0x2230
        github.com/cockroachdb/cockroach/pkg/sql/logictest.(*logicTest).processTestFile(0xc0539d0200, 0xc0037169c0, 0x3f, 0x5319b22, 0xc, 0x3, 0x0, 0x0, 0x0, 0x0, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:1856 +0x7b8
        github.com/cockroachdb/cockroach/pkg/sql/logictest.(*logicTest).runFile(0xc0539d0200, 0xc0037169c0, 0x3f, 0x5319b22, 0xc, 0x3, 0x0, 0x0, 0x0, 0x0, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:3203 +0xd8
        github.com/cockroachdb/cockroach/pkg/sql/logictest.RunLogicTestWithDefaultConfig.func1.1(0xc01fac2c00)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:3409 +0x2de
        testing.tRunner(0xc01fac2c00, 0xc015fa47c0)
        	/usr/local/go/src/testing/testing.go:1193 +0xef
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:1238 +0x2b3
--- done: ../../sql/logictest/testdata/logic_test/distsql_automatic_stats with config 3node-tenant: 10 tests, 0 failures
        --- FAIL: TestTenantLogic/3node-tenant/distsql_automatic_stats (46.81s)
=== RUN   TestTenantLogic/3node-tenant
    --- FAIL: TestTenantLogic/3node-tenant (759.27s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestTenantLogic PKG=./pkg/ccl/logictestccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</details>
/cc @cockroachdb/sql-queries nvanbenschoten
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantLogic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/logictestccl.TestTenantLogic [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3526541&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3526541&tab=artifacts#/) on master @ [7d752a1ecac82d6f44ddb0a3b199f94a6ff85d76](https://github.com/cockroachdb/cockroach/commits/7d752a1ecac82d6f44ddb0a3b199f94a6ff85d76):


```
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTenantLogic887046057
    test_log_scope.go:148: -- test log scope end --
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestTenantLogic
    logic.go:3437: -- test log scope end --
--- FAIL: TestTenantLogic (1155.32s)
=== RUN   TestTenantLogic/3node-tenant/distsql_automatic_stats
    logic.go:2311: condition failed to evaluate within 45s: 
        ../../sql/logictest/testdata/logic_test/distsql_automatic_stats:85: SELECT DISTINCT ON (column_names) statistics_name, column_names, row_count, distinct_count, null_count
        FROM [SHOW STATISTICS FOR TABLE data] ORDER BY column_names ASC, created DESC
        expected:
            statistics_name  column_names  row_count  distinct_count  null_count
            __auto__         {a,b,c}       1050       1050            0
            __auto__         {a,b}         1050       110             0
            __auto__         {a}           1050       11              0
            __auto__         {b}           1050       10              0
            __auto__         {c}           1050       10              0
            __auto__         {d}           1050       3               365
            
        but found (query options: "colnames,rowsort,retry" -> ignore the following ordering of rows) :
            statistics_name  column_names  row_count  distinct_count  null_count
            __auto__         {a}           1000       10              0
            __auto__         {a,b}         1000       100             0
            __auto__         {a,b,c}       1000       1000            0
            __auto__         {b}           1000       10              0
            __auto__         {c}           1000       10              0
            __auto__         {d}           1000       2               730
        goroutine 1735066 [running]:
        runtime/debug.Stack(0xc01f7dd270, 0xa7a358200, 0x6a05360)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0x9f
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithin(0x6ac99d8, 0xc018602780, 0xc01f7dd270, 0xa7a358200)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:60 +0x7b
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x6ac99d8, 0xc018602780, 0xc01f7dd270)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:41 +0x65
        github.com/cockroachdb/cockroach/pkg/sql/logictest.(*logicTest).processSubtest(0xc028d4bc00, 0x0, 0x0, 0xc043a17b30, 0x0, 0xc0080b2a40, 0x3f, 0x530c75c, 0xc, 0x3, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:2311 +0x2230
        github.com/cockroachdb/cockroach/pkg/sql/logictest.(*logicTest).processTestFile(0xc028d4bc00, 0xc0080b2a40, 0x3f, 0x530c75c, 0xc, 0x3, 0x0, 0x0, 0x0, 0x0, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:1856 +0x7b8
        github.com/cockroachdb/cockroach/pkg/sql/logictest.(*logicTest).runFile(0xc028d4bc00, 0xc0080b2a40, 0x3f, 0x530c75c, 0xc, 0x3, 0x0, 0x0, 0x0, 0x0, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:3204 +0xd8
        github.com/cockroachdb/cockroach/pkg/sql/logictest.RunLogicTestWithDefaultConfig.func1.1(0xc018602780)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:3410 +0x2de
        testing.tRunner(0xc018602780, 0xc02fdc27c0)
        	/usr/local/go/src/testing/testing.go:1193 +0xef
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:1238 +0x2b3
--- done: ../../sql/logictest/testdata/logic_test/distsql_automatic_stats with config 3node-tenant: 12 tests, 0 failures
        --- FAIL: TestTenantLogic/3node-tenant/distsql_automatic_stats (48.21s)
=== RUN   TestTenantLogic/3node-tenant
    --- FAIL: TestTenantLogic/3node-tenant (1148.85s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestTenantLogic PKG=./pkg/ccl/logictestccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
/cc @cockroachdb/sql-queries nvanbenschoten
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantLogic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/logictestccl.TestTenantLogic [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3758076&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3758076&tab=artifacts#/) on master @ [f75d6d7ac298c3852f9c5c156b136a8170f483c2](https://github.com/cockroachdb/cockroach/commits/f75d6d7ac298c3852f9c5c156b136a8170f483c2):


```
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTenantLogic1345154637
    test_log_scope.go:148: -- test log scope end --
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestTenantLogic
    logic.go:3514: -- test log scope end --
--- FAIL: TestTenantLogic (1345.10s)
=== RUN   TestTenantLogic/3node-tenant/distsql_automatic_stats
    logic.go:2388: condition failed to evaluate within 45s: 
        ../../sql/logictest/testdata/logic_test/distsql_automatic_stats:66: SELECT DISTINCT ON (column_names) statistics_name, column_names, row_count, distinct_count, null_count
        FROM [SHOW STATISTICS FOR TABLE data] ORDER BY column_names ASC, created DESC
        expected:
            statistics_name  column_names  row_count  distinct_count  null_count
            __auto__         {a,b,c}       1000       1000            0
            __auto__         {a,b}         1000       100             0
            __auto__         {a}           1000       10              0
            __auto__         {b}           1000       10              0
            __auto__         {c}           1000       10              0
            __auto__         {d}           1000       2               730
            
        but found (query options: "colnames,rowsort,retry" -> ignore the following ordering of rows) :
            statistics_name  column_names  row_count  distinct_count  null_count
            __auto__         {a}           1000       10              0
            __auto__         {a,b}         1000       100             0
            __auto__         {a,b,c}       1000       1000            0
            __auto__         {b}           1000       10              0
            __auto__         {c}           1000       10              0
            __auto__         {d}           1000       1               1000
        goroutine 1755481 [running]:
        runtime/debug.Stack()
        	/usr/local/go/src/runtime/debug/stack.go:24 +0x65
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithin({0x61b3fd8, 0xc003ea4000}, 0x0, 0x0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:60 +0x5f
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon({0x61b3fd8, 0xc003ea4000}, 0x3d)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:41 +0x4a
        github.com/cockroachdb/cockroach/pkg/sql/logictest.(*logicTest).processSubtest(0xc0210f2e00, {{0x0, 0x0}, 0xc02271a8a0, 0xc02dc78290}, {0xc020b92a40, 0x0})
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:2388 +0x2954
        github.com/cockroachdb/cockroach/pkg/sql/logictest.(*logicTest).processTestFile(0xc0210f2e00, {0xc020b92a40, 0x15}, {{0x4a4e5fb, 0xc}, 0x3, 0x0, {0x0, 0x0}, {0x0, ...}, ...})
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:1935 +0x5b6
        github.com/cockroachdb/cockroach/pkg/sql/logictest.(*logicTest).runFile(0xc0210f2e00, {0xc020b92a40, 0x7ce6078}, {{0x4a4e5fb, 0xc}, 0x3, 0x0, {0x0, 0x0}, {0x0, ...}, ...})
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:3281 +0xda
        github.com/cockroachdb/cockroach/pkg/sql/logictest.RunLogicTestWithDefaultConfig.func1.1(0xc003ea4000)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:3487 +0x3d0
        testing.tRunner(0xc003ea4000, 0xc006a21540)
        	/usr/local/go/src/testing/testing.go:1259 +0x102
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:1306 +0x35a
--- done: ../../sql/logictest/testdata/logic_test/distsql_automatic_stats with config 3node-tenant: 10 tests, 0 failures
        --- FAIL: TestTenantLogic/3node-tenant/distsql_automatic_stats (48.08s)
=== RUN   TestTenantLogic/3node-tenant
    --- FAIL: TestTenantLogic/3node-tenant (1336.97s)
```
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantLogic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/logictestccl.TestTenantLogic [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3760857&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3760857&tab=artifacts#/) on master @ [28bb1ea049da5bfb6e15a7003cd7b678cbc4b67f](https://github.com/cockroachdb/cockroach/commits/28bb1ea049da5bfb6e15a7003cd7b678cbc4b67f):

Fatal error:

```
panic: pebble: closed

```
Stack: 

```
goroutine 154604774 [running]:
github.com/cockroachdb/pebble.(*DB).newIterInternal(0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/pebble/db.go:746 +0x317
github.com/cockroachdb/pebble.(*DB).NewIter(0x8b2d608, 0xc025face98)
	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/pebble/db.go:923 +0x25
github.com/cockroachdb/cockroach/pkg/storage.(*pebbleIterator).init(0xc01446c240, {0x613fdb0, 0xc021db5b00}, {0x0, 0x0}, {0x1, {0x0, 0x0, 0x0}, {0x0, ...}, ...})
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/pebble_iterator.go:170 +0x6d6
github.com/cockroachdb/cockroach/pkg/storage.newPebbleIterator({0x613fdb0, 0xc021db5b00}, {0x0, 0x0}, {0x1, {0x0, 0x0, 0x0}, {0x0, 0x0, ...}, ...})
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/pebble_iterator.go:90 +0xbb
github.com/cockroachdb/cockroach/pkg/storage.(*Pebble).NewMVCCIterator(0x4518b4, 0x7f707a93ff58, {0x1, {0x0, 0x0, 0x0}, {0x0, 0x0, 0x0}, 0x0, ...})
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/pebble.go:859 +0x185
github.com/cockroachdb/cockroach/pkg/storage.newMVCCIterator(...)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/mvcc.go:826
github.com/cockroachdb/cockroach/pkg/storage.MVCCGet({0x61956f0, 0xc037189ec0}, {0x624a5a8, 0xc024209e60}, {0xc04a6de5e0, 0xa, 0xa}, {0x0, 0x0, 0x0}, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/mvcc.go:856 +0x123
github.com/cockroachdb/cockroach/pkg/server.(*nodeTombstoneStorage).IsDecommissioned(0xc00b686a80, {0x61956f0, 0xc037189ec0}, 0x2)
	/go/src/github.com/cockroachdb/cockroach/pkg/server/node_tombstone_storage.go:62 +0x3d7
github.com/cockroachdb/cockroach/pkg/server.NewServer.func1({0x61956f0, 0xc037189ec0}, 0x1, 0x2)
	/go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:304 +0x71
github.com/cockroachdb/cockroach/pkg/server.NewServer.func3(0x61956f0)
	/go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:338 +0x31
github.com/cockroachdb/cockroach/pkg/rpc.(*HeartbeatService).Ping(0xc0286a7950, {0x61956f0, 0xc02c871440}, 0xc01112bb60)
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/heartbeat.go:172 +0x498
github.com/cockroachdb/cockroach/pkg/rpc._Heartbeat_Ping_Handler.func1({0x61956f0, 0xc02c871440}, {0x48a6580, 0xc01112bb60})
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/heartbeat.pb.go:283 +0x78
github.com/cockroachdb/cockroach/pkg/util/tracing.ServerInterceptor.func1({0x61956f0, 0xc02c871440}, {0x48a6580, 0xc01112bb60}, 0xc033f02720, 0xc039bcf1e8)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/tracing/grpc_interceptor.go:115 +0x234
google.golang.org/grpc.chainUnaryInterceptors.func1.1({0x61956f0, 0xc02c871440}, {0x48a6580, 0xc01112bb60})
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1131 +0x63
github.com/cockroachdb/cockroach/pkg/rpc.NewServer.func1({0x61956f0, 0xc02c871440}, {0x48a6580, 0xc01112bb60}, 0xc02c871410, 0xc02b901480)
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/context.go:183 +0x83
google.golang.org/grpc.chainUnaryInterceptors.func1.1({0x61956f0, 0xc02c871440}, {0x48a6580, 0xc01112bb60})
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1134 +0x87
github.com/cockroachdb/cockroach/pkg/rpc.kvAuth.unaryInterceptor({{{0x10000c00b361b60}}}, {0x61956f0, 0xc02c8713b0}, {0x48a6580, 0xc01112bb60}, 0xc033f02720, 0xc02b901480)
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/auth.go:71 +0x173
google.golang.org/grpc.chainUnaryInterceptors.func1.1({0x61956f0, 0xc02c8713b0}, {0x48a6580, 0xc01112bb60})
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1134 +0x87
google.golang.org/grpc.chainUnaryInterceptors.func1({0x61956f0, 0xc02c8713b0}, {0x48a6580, 0xc01112bb60}, 0xc033f02720, 0xc039bcf1e8)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1136 +0x159
github.com/cockroachdb/cockroach/pkg/rpc._Heartbeat_Ping_Handler({0x438f3c0, 0xc0286a7950}, {0x61956f0, 0xc02c8713b0}, 0xc01112bb00, 0xc022f3bc80)
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/heartbeat.pb.go:285 +0x138
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00fcf7180, {0x621ea18, 0xc0156f6d80}, 0xc00e46c900, 0xc02ed06a80, 0x8a4e030, 0x0)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1297 +0xccf
google.golang.org/grpc.(*Server).handleStream(0xc00fcf7180, {0x621ea18, 0xc0156f6d80}, 0xc00e46c900, 0x0)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1626 +0xa2a
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:941 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:939 +0x294
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestTenantLogic/3node-tenant/postgresjoin
--- progress: ../../sql/logictest/testdata/logic_test/postgresjoin: 101 statements/queries
--- done: ../../sql/logictest/testdata/logic_test/postgresjoin with config 3node-tenant: 101 tests, 0 failures
```
</p>
</details>
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)Parameters in this failure:

- TAGS=deadlock

- GOFLAGS=-parallel=4
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantLogic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/logictestccl.TestTenantLogic [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3781610&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3781610&tab=artifacts#/) on master @ [a5e32e1d18aa07a65eb44063177a6c196623f360](https://github.com/cockroachdb/cockroach/commits/a5e32e1d18aa07a65eb44063177a6c196623f360):

Fatal error:

```
panic: pebble: closed

```
Stack: 

```
goroutine 108040715 [running]:
github.com/cockroachdb/pebble.(*DB).newIterInternal(0xc0066ea300, 0x0, 0x8, 0xc0114bee60)
	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/pebble/db.go:746 +0x317
github.com/cockroachdb/pebble.(*DB).NewIter(0x0, 0x451800)
	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/pebble/db.go:923 +0x25
github.com/cockroachdb/cockroach/pkg/storage.(*pebbleIterator).init(0xc00f4deb40, {0x614a780, 0xc01cf82480}, {0x0, 0x0}, {0x1, {0x0, 0x0, 0x0}, {0x0, ...}, ...})
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/pebble_iterator.go:170 +0x6d6
github.com/cockroachdb/cockroach/pkg/storage.newPebbleIterator({0x614a780, 0xc01cf82480}, {0x0, 0x0}, {0x1, {0x0, 0x0, 0x0}, {0x0, 0x0, ...}, ...})
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/pebble_iterator.go:90 +0xbb
github.com/cockroachdb/cockroach/pkg/storage.(*Pebble).NewMVCCIterator(0x4518b4, 0xc0114bf1b0, {0x1, {0x0, 0x0, 0x0}, {0x0, 0x0, 0x0}, 0x0, ...})
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/pebble.go:859 +0x185
github.com/cockroachdb/cockroach/pkg/storage.newMVCCIterator(...)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/mvcc.go:826
github.com/cockroachdb/cockroach/pkg/storage.MVCCGet({0x61a0090, 0xc025413e60}, {0x6255088, 0xc01ca510e0}, {0xc010318050, 0xa, 0xa}, {0x0, 0x0, 0x0}, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/mvcc.go:856 +0x123
github.com/cockroachdb/cockroach/pkg/server.(*nodeTombstoneStorage).IsDecommissioned(0xc03036bc40, {0x61a0090, 0xc025413e60}, 0x3)
	/go/src/github.com/cockroachdb/cockroach/pkg/server/node_tombstone_storage.go:62 +0x3d7
github.com/cockroachdb/cockroach/pkg/server.NewServer.func1({0x61a0090, 0xc025413e60}, 0x1, 0x2)
	/go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:304 +0x71
github.com/cockroachdb/cockroach/pkg/server.NewServer.func3(0x61a0090)
	/go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:338 +0x31
github.com/cockroachdb/cockroach/pkg/rpc.(*HeartbeatService).Ping(0xc0355dbb80, {0x61a0090, 0xc02cb36c30}, 0xc00130fda0)
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/heartbeat.go:172 +0x498
github.com/cockroachdb/cockroach/pkg/rpc._Heartbeat_Ping_Handler.func1({0x61a0090, 0xc02cb36c30}, {0x48aed20, 0xc00130fda0})
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/heartbeat.pb.go:283 +0x78
github.com/cockroachdb/cockroach/pkg/util/tracing.ServerInterceptor.func1({0x61a0090, 0xc02cb36c30}, {0x48aed20, 0xc00130fda0}, 0xc02c955e00, 0xc0332b39b0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/tracing/grpc_interceptor.go:115 +0x234
google.golang.org/grpc.chainUnaryInterceptors.func1.1({0x61a0090, 0xc02cb36c30}, {0x48aed20, 0xc00130fda0})
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1131 +0x63
github.com/cockroachdb/cockroach/pkg/rpc.NewServer.func1({0x61a0090, 0xc02cb36c30}, {0x48aed20, 0xc00130fda0}, 0xc02cb36c00, 0xc00208a100)
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/context.go:183 +0x83
google.golang.org/grpc.chainUnaryInterceptors.func1.1({0x61a0090, 0xc02cb36c30}, {0x48aed20, 0xc00130fda0})
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1134 +0x87
github.com/cockroachdb/cockroach/pkg/rpc.kvAuth.unaryInterceptor({{{0x10000c000283b60}}}, {0x61a0090, 0xc0020663c0}, {0x48aed20, 0xc00130fda0}, 0xc02c955e00, 0xc00208a100)
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/auth.go:71 +0x173
google.golang.org/grpc.chainUnaryInterceptors.func1.1({0x61a0090, 0xc0020663c0}, {0x48aed20, 0xc00130fda0})
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1134 +0x87
google.golang.org/grpc.chainUnaryInterceptors.func1({0x61a0090, 0xc0020663c0}, {0x48aed20, 0xc00130fda0}, 0xc02c955e00, 0xc0332b39b0)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1136 +0x159
github.com/cockroachdb/cockroach/pkg/rpc._Heartbeat_Ping_Handler({0x4396b80, 0xc0355dbb80}, {0x61a0090, 0xc0020663c0}, 0xc00130fd40, 0xc01b228cc0)
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/heartbeat.pb.go:285 +0x138
google.golang.org/grpc.(*Server).processUnaryRPC(0xc017e6f180, {0x62294f8, 0xc02564ac00}, 0xc02c97ed80, 0xc0224ae240, 0x8a5d030, 0x0)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1297 +0xccf
google.golang.org/grpc.(*Server).handleStream(0xc017e6f180, {0x62294f8, 0xc02564ac00}, 0xc02c97ed80, 0x0)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1626 +0xa2a
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:941 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:939 +0x294
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestTenantLogic/3node-tenant/datetime/datestyle_order
--- done: ../../sql/logictest/testdata/logic_test/datetime with config 3node-tenant: 329 tests, 0 failures
```
</p>
</details>
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)Parameters in this failure:

- TAGS=deadlock

- GOFLAGS=-parallel=4
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantLogic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/logictestccl.TestTenantLogic [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3807457&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3807457&tab=artifacts#/) on master @ [3b30a0e12f9a14b08ee8ad55b50299aca50c67a2](https://github.com/cockroachdb/cockroach/commits/3b30a0e12f9a14b08ee8ad55b50299aca50c67a2):

Fatal error:

```
panic: pebble: closed

```
Stack: 

```
goroutine 156514109 [running]:
github.com/cockroachdb/pebble.(*DB).newIterInternal(0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/pebble/db.go:746 +0x317
github.com/cockroachdb/pebble.(*DB).NewIter(0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/pebble/db.go:923 +0x25
github.com/cockroachdb/cockroach/pkg/storage.(*pebbleIterator).init(0xc00c6738c0, {0x61b3c50, 0xc02522a480}, {0x0, 0x0}, {0x1, {0x0, 0x0, 0x0}, {0x0, ...}, ...})
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/pebble_iterator.go:170 +0x6d6
github.com/cockroachdb/cockroach/pkg/storage.newPebbleIterator({0x61b3c50, 0xc02522a480}, {0x0, 0x0}, {0x1, {0x0, 0x0, 0x0}, {0x0, 0x0, ...}, ...})
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/pebble_iterator.go:90 +0xbb
github.com/cockroachdb/cockroach/pkg/storage.(*Pebble).NewMVCCIterator(0x4518b4, 0x10, {0x1, {0x0, 0x0, 0x0}, {0x0, 0x0, 0x0}, 0x0, ...})
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/pebble.go:859 +0x185
github.com/cockroachdb/cockroach/pkg/storage.newMVCCIterator(...)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/mvcc.go:826
github.com/cockroachdb/cockroach/pkg/storage.MVCCGet({0x6209d10, 0xc01c556900}, {0x62bfc40, 0xc01b44b9e0}, {0xc037816960, 0xa, 0xa}, {0x0, 0x0, 0x0}, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/mvcc.go:856 +0x123
github.com/cockroachdb/cockroach/pkg/server.(*nodeTombstoneStorage).IsDecommissioned(0xc00fcf1dc0, {0x6209d10, 0xc01c556900}, 0x3)
	/go/src/github.com/cockroachdb/cockroach/pkg/server/node_tombstone_storage.go:62 +0x3d7
github.com/cockroachdb/cockroach/pkg/server.NewServer.func1({0x6209d10, 0xc01c556900}, 0x1, 0x2)
	/go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:308 +0x71
github.com/cockroachdb/cockroach/pkg/server.NewServer.func3(0x6209d10)
	/go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:342 +0x31
github.com/cockroachdb/cockroach/pkg/rpc.(*HeartbeatService).Ping(0xc0085e4870, {0x6209d10, 0xc035641800}, 0xc0185db800)
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/heartbeat.go:172 +0x498
github.com/cockroachdb/cockroach/pkg/rpc._Heartbeat_Ping_Handler.func1({0x6209d10, 0xc035641800}, {0x49082a0, 0xc0185db800})
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/heartbeat.pb.go:283 +0x78
github.com/cockroachdb/cockroach/pkg/util/tracing.ServerInterceptor.func1({0x6209d10, 0xc035641800}, {0x49082a0, 0xc0185db800}, 0xc037e980a0, 0xc013fbefd8)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/tracing/grpc_interceptor.go:115 +0x234
google.golang.org/grpc.chainUnaryInterceptors.func1.1({0x6209d10, 0xc035641800}, {0x49082a0, 0xc0185db800})
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1131 +0x63
github.com/cockroachdb/cockroach/pkg/rpc.NewServer.func1({0x6209d10, 0xc035641800}, {0x49082a0, 0xc0185db800}, 0xc0356417d0, 0xc030c1d3c0)
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/context.go:183 +0x83
google.golang.org/grpc.chainUnaryInterceptors.func1.1({0x6209d10, 0xc035641800}, {0x49082a0, 0xc0185db800})
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1134 +0x87
github.com/cockroachdb/cockroach/pkg/rpc.kvAuth.unaryInterceptor({{{0x10000c02e110b60}}}, {0x6209d10, 0xc035641770}, {0x49082a0, 0xc0185db800}, 0xc037e980a0, 0xc030c1d3c0)
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/auth.go:71 +0x173
google.golang.org/grpc.chainUnaryInterceptors.func1.1({0x6209d10, 0xc035641770}, {0x49082a0, 0xc0185db800})
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1134 +0x87
google.golang.org/grpc.chainUnaryInterceptors.func1({0x6209d10, 0xc035641770}, {0x49082a0, 0xc0185db800}, 0xc037e980a0, 0xc013fbefd8)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1136 +0x159
github.com/cockroachdb/cockroach/pkg/rpc._Heartbeat_Ping_Handler({0x43e8d80, 0xc0085e4870}, {0x6209d10, 0xc035641770}, 0xc0185db7a0, 0xc0004eb080)
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/heartbeat.pb.go:285 +0x138
google.golang.org/grpc.(*Server).processUnaryRPC(0xc01d79e700, {0x6293e98, 0xc02d5c2900}, 0xc0207bc120, 0xc023928600, 0x8af9040, 0x0)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1297 +0xccf
google.golang.org/grpc.(*Server).handleStream(0xc01d79e700, {0x6293e98, 0xc02d5c2900}, 0xc0207bc120, 0x0)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1626 +0xa2a
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:941 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:939 +0x294
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestTenantLogic/3node-tenant/order_by/nulls_ordering
--- done: ../../sql/logictest/testdata/logic_test/order_by with config 3node-tenant: 92 tests, 0 failures
```
</p>
</details>
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)Parameters in this failure:

- TAGS=deadlock

- GOFLAGS=-parallel=4
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantLogic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/logictestccl.TestTenantLogic [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3810397&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3810397&tab=artifacts#/) on master @ [2c014c47c1a242f504f6d595bfd79c0edc20b90a](https://github.com/cockroachdb/cockroach/commits/2c014c47c1a242f504f6d595bfd79c0edc20b90a):

Fatal error:

```
panic: pebble: closed

```
Stack: 

```
goroutine 1313835 [running]:
github.com/cockroachdb/pebble.(*DB).newIterInternal(0xc009e80000, 0x3, 0xc013a28e38, 0x44a289)
	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/pebble/db.go:746 +0x317
github.com/cockroachdb/pebble.(*DB).NewIter(0xc013a29168, 0xc009e80000)
	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/pebble/db.go:923 +0x25
github.com/cockroachdb/cockroach/pkg/storage.(*pebbleIterator).init(0xc011eb78c0, {0x61af3b0, 0xc00ca3f680}, {0x0, 0x0}, {0x1, {0x0, 0x0, 0x0}, {0x0, ...}, ...})
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/pebble_iterator.go:170 +0x6d6
github.com/cockroachdb/cockroach/pkg/storage.newPebbleIterator({0x61af3b0, 0xc00ca3f680}, {0x0, 0x0}, {0x1, {0x0, 0x0, 0x0}, {0x0, 0x0, ...}, ...})
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/pebble_iterator.go:90 +0xbb
github.com/cockroachdb/cockroach/pkg/storage.(*Pebble).NewMVCCIterator(0xc013a291d0, 0x4d0654, {0x1, {0x0, 0x0, 0x0}, {0x0, 0x0, 0x0}, 0x0, ...})
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/pebble.go:859 +0x185
github.com/cockroachdb/cockroach/pkg/storage.newMVCCIterator(...)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/mvcc.go:826
github.com/cockroachdb/cockroach/pkg/storage.MVCCGet({0x61ff1f0, 0xc00f925a40}, {0x62b1300, 0xc006f56240}, {0xc0060def60, 0xa, 0xa}, {0x0, 0x0, 0x0}, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/mvcc.go:856 +0x123
github.com/cockroachdb/cockroach/pkg/server.(*nodeTombstoneStorage).IsDecommissioned(0xc018ae51c0, {0x61ff1f0, 0xc00f925a40}, 0x2)
	/go/src/github.com/cockroachdb/cockroach/pkg/server/node_tombstone_storage.go:62 +0x3e8
github.com/cockroachdb/cockroach/pkg/server.NewServer.func1({0x61ff1f0, 0xc00f925a40}, 0x1, 0x2)
	/go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:308 +0x71
github.com/cockroachdb/cockroach/pkg/server.NewServer.func3(0x61ff1f0)
	/go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:342 +0x31
github.com/cockroachdb/cockroach/pkg/rpc.(*HeartbeatService).Ping(0xc00f6cbf90, {0x61ff1f0, 0xc0291b4a80}, 0xc00ba15d40)
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/heartbeat.go:172 +0x498
github.com/cockroachdb/cockroach/pkg/rpc._Heartbeat_Ping_Handler.func1({0x61ff1f0, 0xc0291b4a80}, {0x4900480, 0xc00ba15d40})
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/heartbeat.pb.go:283 +0x78
github.com/cockroachdb/cockroach/pkg/util/tracing.ServerInterceptor.func1({0x61ff1f0, 0xc0291b4a80}, {0x4900480, 0xc00ba15d40}, 0xc004230620, 0xc00ee17848)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/tracing/grpc_interceptor.go:115 +0x234
google.golang.org/grpc.chainUnaryInterceptors.func1.1({0x61ff1f0, 0xc0291b4a80}, {0x4900480, 0xc00ba15d40})
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1131 +0x63
github.com/cockroachdb/cockroach/pkg/rpc.NewServer.func1({0x61ff1f0, 0xc0291b4a80}, {0x4900480, 0xc00ba15d40}, 0xc0291b4a50, 0xc03b204900)
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/context.go:183 +0x83
google.golang.org/grpc.chainUnaryInterceptors.func1.1({0x61ff1f0, 0xc0291b4a80}, {0x4900480, 0xc00ba15d40})
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1134 +0x87
github.com/cockroachdb/cockroach/pkg/rpc.kvAuth.unaryInterceptor({{{0x10000c007f03b60}}}, {0x61ff1f0, 0xc0291b49c0}, {0x4900480, 0xc00ba15d40}, 0xc004230620, 0xc03b204900)
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/auth.go:71 +0x173
google.golang.org/grpc.chainUnaryInterceptors.func1.1({0x61ff1f0, 0xc0291b49c0}, {0x4900480, 0xc00ba15d40})
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1134 +0x87
google.golang.org/grpc.chainUnaryInterceptors.func1({0x61ff1f0, 0xc0291b49c0}, {0x4900480, 0xc00ba15d40}, 0xc004230620, 0xc00ee17848)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1136 +0x159
github.com/cockroachdb/cockroach/pkg/rpc._Heartbeat_Ping_Handler({0x43e0800, 0xc00f6cbf90}, {0x61ff1f0, 0xc0291b49c0}, 0xc00ba15ce0, 0xc00dec0620)
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/heartbeat.pb.go:285 +0x138
google.golang.org/grpc.(*Server).processUnaryRPC(0xc010bf2380, {0x6287158, 0xc0129b7800}, 0xc01512bd40, 0xc02261ec90, 0x8ae9040, 0x0)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1297 +0xccf
google.golang.org/grpc.(*Server).handleStream(0xc010bf2380, {0x6287158, 0xc0129b7800}, 0xc01512bd40, 0x0)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1626 +0xa2a
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:941 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:939 +0x294
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestTenantLogic/3node-tenant/create_table/generated_as_identity_with_seq_option
--- done: ../../sql/logictest/testdata/logic_test/create_table with config 3node-tenant: 95 tests, 0 failures
```
</p>
</details>
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantLogic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/logictestccl.TestTenantLogic [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3844704&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3844704&tab=artifacts#/) on master @ [506d129f5f187134c35e2f71860490e044fde989](https://github.com/cockroachdb/cockroach/commits/506d129f5f187134c35e2f71860490e044fde989):

Fatal error:

```
panic: pebble: closed

```
Stack: 

```
goroutine 2902771 [running]:
github.com/cockroachdb/pebble.(*DB).newIterInternal(0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/pebble/db.go:746 +0x317
github.com/cockroachdb/pebble.(*DB).NewIter(0xc01370ce88, 0x4518b4)
	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/pebble/db.go:923 +0x25
github.com/cockroachdb/cockroach/pkg/storage.(*pebbleIterator).init(0xc0067bd0e0, {0x61d5080, 0xc009327b00}, {0x0, 0x0}, {0x1, {0x0, 0x0, 0x0}, {0x0, ...}, ...})
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/pebble_iterator.go:170 +0x6d6
github.com/cockroachdb/cockroach/pkg/storage.newPebbleIterator({0x61d5080, 0xc009327b00}, {0x0, 0x0}, {0x1, {0x0, 0x0, 0x0}, {0x0, 0x0, ...}, ...})
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/pebble_iterator.go:90 +0xbb
github.com/cockroachdb/cockroach/pkg/storage.(*Pebble).NewMVCCIterator(0x0, 0x4, {0x1, {0x0, 0x0, 0x0}, {0x0, 0x0, 0x0}, 0x0, ...})
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/pebble.go:855 +0x185
github.com/cockroachdb/cockroach/pkg/storage.newMVCCIterator(...)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/mvcc.go:804
github.com/cockroachdb/cockroach/pkg/storage.MVCCGet({0x6224f30, 0xc00da9cc60}, {0x62d74a0, 0xc00f13fd00}, {0xc0079d1300, 0xa, 0xa}, {0x0, 0x0, 0x0}, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/mvcc.go:834 +0x123
github.com/cockroachdb/cockroach/pkg/server.(*nodeTombstoneStorage).IsDecommissioned(0xc019053ac0, {0x6224f30, 0xc00da9cc60}, 0x2)
	/go/src/github.com/cockroachdb/cockroach/pkg/server/node_tombstone_storage.go:62 +0x3f1
github.com/cockroachdb/cockroach/pkg/server.NewServer.func1({0x6224f30, 0xc00da9cc60}, 0x1, 0x2)
	/go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:309 +0x71
github.com/cockroachdb/cockroach/pkg/server.NewServer.func3(0x6224f30)
	/go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:343 +0x31
github.com/cockroachdb/cockroach/pkg/rpc.(*HeartbeatService).Ping(0xc0395a87d0, {0x6224f30, 0xc006ce8810}, 0xc011d60ba0)
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/heartbeat.go:172 +0x498
github.com/cockroachdb/cockroach/pkg/rpc._Heartbeat_Ping_Handler.func1({0x6224f30, 0xc006ce8810}, {0x4916e60, 0xc011d60ba0})
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/heartbeat.pb.go:283 +0x78
github.com/cockroachdb/cockroach/pkg/util/tracing.ServerInterceptor.func1({0x6224f30, 0xc006ce8810}, {0x4916e60, 0xc011d60ba0}, 0xc010e17540, 0xc03237fc68)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/tracing/grpc_interceptor.go:115 +0x234
google.golang.org/grpc.chainUnaryInterceptors.func1.1({0x6224f30, 0xc006ce8810}, {0x4916e60, 0xc011d60ba0})
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1131 +0x63
github.com/cockroachdb/cockroach/pkg/rpc.NewServer.func1({0x6224f30, 0xc006ce8810}, {0x4916e60, 0xc011d60ba0}, 0xc006ce87e0, 0xc009728880)
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/context.go:183 +0x83
google.golang.org/grpc.chainUnaryInterceptors.func1.1({0x6224f30, 0xc006ce8810}, {0x4916e60, 0xc011d60ba0})
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1134 +0x87
github.com/cockroachdb/cockroach/pkg/rpc.kvAuth.unaryInterceptor({{{0x10000c00645bb60}}}, {0x6224f30, 0xc006ce8780}, {0x4916e60, 0xc011d60ba0}, 0xc010e17540, 0xc009728880)
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/auth.go:71 +0x173
google.golang.org/grpc.chainUnaryInterceptors.func1.1({0x6224f30, 0xc006ce8780}, {0x4916e60, 0xc011d60ba0})
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1134 +0x87
google.golang.org/grpc.chainUnaryInterceptors.func1({0x6224f30, 0xc006ce8780}, {0x4916e60, 0xc011d60ba0}, 0xc010e17540, 0xc03237fc68)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1136 +0x159
github.com/cockroachdb/cockroach/pkg/rpc._Heartbeat_Ping_Handler({0x43f5b60, 0xc0395a87d0}, {0x6224f30, 0xc006ce8780}, 0xc011d60b40, 0xc00e66e620)
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/heartbeat.pb.go:285 +0x138
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0190ab880, {0x62ace80, 0xc003a2b080}, 0xc004f645a0, 0xc015255290, 0x8b1c060, 0x0)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1297 +0xccf
google.golang.org/grpc.(*Server).handleStream(0xc0190ab880, {0x62ace80, 0xc003a2b080}, 0xc004f645a0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1626 +0xa2a
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:941 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:939 +0x294
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestTenantLogic/3node-tenant/storing
--- progress: ../../sql/logictest/testdata/logic_test/storing: 33 statements/queries
--- done: ../../sql/logictest/testdata/logic_test/storing with config 3node-tenant: 41 tests, 0 failures
```
</p>
</details>
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantLogic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/logictestccl.TestTenantLogic [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3860549&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3860549&tab=artifacts#/) on master @ [39923c0b11d229b394fa5498ee58e455cae8ec99](https://github.com/cockroachdb/cockroach/commits/39923c0b11d229b394fa5498ee58e455cae8ec99):

Fatal error:

```
panic: pebble: closed

```
Stack: 

```
goroutine 2904325 [running]:
github.com/cockroachdb/pebble.(*DB).newIterInternal(0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/pebble/db.go:746 +0x317
github.com/cockroachdb/pebble.(*DB).NewIter(0x0, 0x984130acb4e7ccda)
	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/pebble/db.go:923 +0x25
github.com/cockroachdb/cockroach/pkg/storage.(*pebbleIterator).init(0xc022cd2360, {0x61e5af0, 0xc014f3ad80}, {0x0, 0x0}, {0x1, {0x0, 0x0, 0x0}, {0x0, ...}, ...})
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/pebble_iterator.go:179 +0x7b6
github.com/cockroachdb/cockroach/pkg/storage.newPebbleIterator({0x61e5af0, 0xc014f3ad80}, {0x0, 0x0}, {0x1, {0x0, 0x0, 0x0}, {0x0, 0x0, ...}, ...})
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/pebble_iterator.go:91 +0xbb
github.com/cockroachdb/cockroach/pkg/storage.(*Pebble).NewMVCCIterator(0x10000c003935db0, 0x7f4ea9abd120, {0x1, {0x0, 0x0, 0x0}, {0x0, 0x0, 0x0}, 0x0, ...})
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/pebble.go:965 +0x185
github.com/cockroachdb/cockroach/pkg/storage.newMVCCIterator(...)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/mvcc.go:804
github.com/cockroachdb/cockroach/pkg/storage.MVCCGet({0x6235ab0, 0xc02247ba70}, {0x62e8100, 0xc00cb7ec00}, {0xc002199a20, 0xa, 0xa}, {0x0, 0x0, 0x0}, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/mvcc.go:834 +0x123
github.com/cockroachdb/cockroach/pkg/server.(*nodeTombstoneStorage).IsDecommissioned(0xc01146d4c0, {0x6235ab0, 0xc02247ba70}, 0x2)
	/go/src/github.com/cockroachdb/cockroach/pkg/server/node_tombstone_storage.go:62 +0x3f1
github.com/cockroachdb/cockroach/pkg/server.NewServer.func1({0x6235ab0, 0xc02247ba70}, 0x1, 0x2)
	/go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:309 +0x71
github.com/cockroachdb/cockroach/pkg/server.NewServer.func3(0x6235ab0)
	/go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:343 +0x31
github.com/cockroachdb/cockroach/pkg/rpc.(*HeartbeatService).Ping(0xc01ca90aa0, {0x6235ab0, 0xc010fed950}, 0xc0453ad320)
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/heartbeat.go:172 +0x498
github.com/cockroachdb/cockroach/pkg/rpc._Heartbeat_Ping_Handler.func1({0x6235ab0, 0xc010fed950}, {0x491c2e0, 0xc0453ad320})
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/heartbeat.pb.go:283 +0x78
github.com/cockroachdb/cockroach/pkg/util/tracing.ServerInterceptor.func1({0x6235ab0, 0xc010fed950}, {0x491c2e0, 0xc0453ad320}, 0xc01fae7e40, 0xc0206a1518)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/tracing/grpc_interceptor.go:115 +0x234
google.golang.org/grpc.chainUnaryInterceptors.func1.1({0x6235ab0, 0xc010fed950}, {0x491c2e0, 0xc0453ad320})
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1131 +0x63
github.com/cockroachdb/cockroach/pkg/rpc.NewServer.func1({0x6235ab0, 0xc010fed950}, {0x491c2e0, 0xc0453ad320}, 0xc010fed920, 0xc011015540)
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/context.go:183 +0x83
google.golang.org/grpc.chainUnaryInterceptors.func1.1({0x6235ab0, 0xc010fed950}, {0x491c2e0, 0xc0453ad320})
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1134 +0x87
github.com/cockroachdb/cockroach/pkg/rpc.kvAuth.unaryInterceptor({{{0x10000c0101f4b60}}}, {0x6235ab0, 0xc010fed8c0}, {0x491c2e0, 0xc0453ad320}, 0xc01fae7e40, 0xc011015540)
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/auth.go:71 +0x173
google.golang.org/grpc.chainUnaryInterceptors.func1.1({0x6235ab0, 0xc010fed8c0}, {0x491c2e0, 0xc0453ad320})
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1134 +0x87
google.golang.org/grpc.chainUnaryInterceptors.func1({0x6235ab0, 0xc010fed8c0}, {0x491c2e0, 0xc0453ad320}, 0xc01fae7e40, 0xc0206a1518)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1136 +0x159
github.com/cockroachdb/cockroach/pkg/rpc._Heartbeat_Ping_Handler({0x43fa0a0, 0xc01ca90aa0}, {0x6235ab0, 0xc010fed8c0}, 0xc0453ad2c0, 0xc0218e0be0)
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/heartbeat.pb.go:285 +0x138
google.golang.org/grpc.(*Server).processUnaryRPC(0xc02b738540, {0x62bda60, 0xc012f05380}, 0xc0110130e0, 0xc029872150, 0x8b2e070, 0x0)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1297 +0xccf
google.golang.org/grpc.(*Server).handleStream(0xc02b738540, {0x62bda60, 0xc012f05380}, 0xc0110130e0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1626 +0xa2a
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:941 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:939 +0x294
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestTenantLogic/3node-tenant/storing
--- progress: ../../sql/logictest/testdata/logic_test/storing: 33 statements/queries
--- done: ../../sql/logictest/testdata/logic_test/storing with config 3node-tenant: 41 tests, 0 failures
```
</p>
</details>
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantLogic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/logictestccl.TestTenantLogic [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3874682&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3874682&tab=artifacts#/) on master @ [e89328d92398a3e2d6487179845a51e7f1caa435](https://github.com/cockroachdb/cockroach/commits/e89328d92398a3e2d6487179845a51e7f1caa435):

Fatal error:

```
panic: pebble: closed

```
Stack: 

```
goroutine 2913881 [running]:
github.com/cockroachdb/pebble.(*DB).newIterInternal(0xc01f511bc0, 0x100000001, 0x3, 0xc008fa14d6)
	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/pebble/db.go:746 +0x317
github.com/cockroachdb/pebble.(*DB).NewIter(0x100000001, 0x4)
	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/pebble/db.go:923 +0x25
github.com/cockroachdb/cockroach/pkg/storage.(*pebbleIterator).init(0xc013f90480, {0x61f2fd0, 0xc0193d3680}, {0x0, 0x0}, {0x1, {0x0, 0x0, 0x0}, {0x0, ...}, ...})
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/pebble_iterator.go:179 +0x7b6
github.com/cockroachdb/cockroach/pkg/storage.newPebbleIterator({0x61f2fd0, 0xc0193d3680}, {0x0, 0x0}, {0x1, {0x0, 0x0, 0x0}, {0x0, 0x0, ...}, ...})
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/pebble_iterator.go:91 +0xbb
github.com/cockroachdb/cockroach/pkg/storage.(*Pebble).NewMVCCIterator(0x0, 0x0, {0x1, {0x0, 0x0, 0x0}, {0x0, 0x0, 0x0}, 0x0, ...})
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/pebble.go:965 +0x185
github.com/cockroachdb/cockroach/pkg/storage.newMVCCIterator(...)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/mvcc.go:804
github.com/cockroachdb/cockroach/pkg/storage.MVCCGet({0x62430d0, 0xc028b8b0b0}, {0x62f5740, 0xc01cc36a00}, {0xc0131fa010, 0xa, 0xa}, {0x0, 0x0, 0x0}, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/mvcc.go:834 +0x123
github.com/cockroachdb/cockroach/pkg/server.(*nodeTombstoneStorage).IsDecommissioned(0xc02092f280, {0x62430d0, 0xc028b8b0b0}, 0x2)
	/go/src/github.com/cockroachdb/cockroach/pkg/server/node_tombstone_storage.go:62 +0x3f1
github.com/cockroachdb/cockroach/pkg/server.NewServer.func1({0x62430d0, 0xc028b8b0b0}, 0x1, 0x2)
	/go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:307 +0x71
github.com/cockroachdb/cockroach/pkg/server.NewServer.func3(0x62430d0)
	/go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:343 +0x31
github.com/cockroachdb/cockroach/pkg/rpc.(*HeartbeatService).Ping(0xc0135c2be0, {0x62430d0, 0xc018ca30e0}, 0xc020be8d20)
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/heartbeat.go:172 +0x498
github.com/cockroachdb/cockroach/pkg/rpc._Heartbeat_Ping_Handler.func1({0x62430d0, 0xc018ca30e0}, {0x4927980, 0xc020be8d20})
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/heartbeat.pb.go:283 +0x78
github.com/cockroachdb/cockroach/pkg/util/tracing.ServerInterceptor.func1({0x62430d0, 0xc018ca30e0}, {0x4927980, 0xc020be8d20}, 0xc036568cc0, 0xc067a36840)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/tracing/grpc_interceptor.go:115 +0x234
google.golang.org/grpc.chainUnaryInterceptors.func1.1({0x62430d0, 0xc018ca30e0}, {0x4927980, 0xc020be8d20})
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1131 +0x63
github.com/cockroachdb/cockroach/pkg/rpc.NewServer.func1({0x62430d0, 0xc018ca30e0}, {0x4927980, 0xc020be8d20}, 0xc018ca30b0, 0xc018c99500)
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/context.go:183 +0x83
google.golang.org/grpc.chainUnaryInterceptors.func1.1({0x62430d0, 0xc018ca30e0}, {0x4927980, 0xc020be8d20})
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1134 +0x87
github.com/cockroachdb/cockroach/pkg/rpc.kvAuth.unaryInterceptor({{{0x10000c0069c0b60}}}, {0x62430d0, 0xc018ca3050}, {0x4927980, 0xc020be8d20}, 0xc036568cc0, 0xc018c99500)
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/auth.go:71 +0x173
google.golang.org/grpc.chainUnaryInterceptors.func1.1({0x62430d0, 0xc018ca3050}, {0x4927980, 0xc020be8d20})
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1134 +0x87
google.golang.org/grpc.chainUnaryInterceptors.func1({0x62430d0, 0xc018ca3050}, {0x4927980, 0xc020be8d20}, 0xc036568cc0, 0xc067a36840)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1136 +0x159
github.com/cockroachdb/cockroach/pkg/rpc._Heartbeat_Ping_Handler({0x4404e00, 0xc0135c2be0}, {0x62430d0, 0xc018ca3050}, 0xc020be8cc0, 0xc04d00c300)
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/heartbeat.pb.go:285 +0x138
google.golang.org/grpc.(*Server).processUnaryRPC(0xc027afe1c0, {0x62caf90, 0xc00ed14900}, 0xc018ba5e60, 0xc029ea5e30, 0x8b42070, 0x0)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1297 +0xccf
google.golang.org/grpc.(*Server).handleStream(0xc027afe1c0, {0x62caf90, 0xc00ed14900}, 0xc018ba5e60, 0x0)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1626 +0xa2a
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:941 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:939 +0x294
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestTenantLogic/3node-tenant/storing
--- progress: ../../sql/logictest/testdata/logic_test/storing: 33 statements/queries
--- done: ../../sql/logictest/testdata/logic_test/storing with config 3node-tenant: 41 tests, 0 failures
```
</p>
</details>
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #73615 ccl/logictestccl: TestTenantLogic failed [C-test-failure O-robot T-sql-queries branch-release-21.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantLogic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/logictestccl.TestTenantLogic [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3975266&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3975266&tab=artifacts#/) on master @ [4c1b9fb7ac9e058111cebacdb4d98c9ebacf6bf7](https://github.com/cockroachdb/cockroach/commits/4c1b9fb7ac9e058111cebacdb4d98c9ebacf6bf7):


```
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTenantLogic681008792
    test_log_scope.go:148: -- test log scope end --
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestTenantLogic
    logic.go:3593: -- test log scope end --
--- FAIL: TestTenantLogic (991.83s)
=== RUN   TestTenantLogic/3node-tenant/distsql_automatic_stats
    logic.go:2432: condition failed to evaluate within 45s: 
        ../../sql/logictest/testdata/logic_test/distsql_automatic_stats:66: SELECT DISTINCT ON (column_names) statistics_name, column_names, row_count, distinct_count, null_count
        FROM [SHOW STATISTICS FOR TABLE data] ORDER BY column_names ASC, created DESC
        expected:
            statistics_name  column_names  row_count  distinct_count  null_count
            __auto__         {a,b,c}       1000       1000            0
            __auto__         {a,b}         1000       100             0
            __auto__         {a}           1000       10              0
            __auto__         {b}           1000       10              0
            __auto__         {c}           1000       10              0
            __auto__         {d}           1000       2               730
            
        but found (query options: "colnames,rowsort,retry" -> ignore the following ordering of rows) :
            statistics_name  column_names  row_count  distinct_count  null_count
            __auto__         {a}           1000       10              0
            __auto__         {a,b}         1000       100             0
            __auto__         {a,b,c}       1000       1000            0
            __auto__         {b}           1000       10              0
            __auto__         {c}           1000       10              0
            __auto__         {d}           1000       1               1000
        goroutine 1617807 [running]:
        runtime/debug.Stack()
        	/usr/local/go/src/runtime/debug/stack.go:24 +0x65
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithin({0x5d3e7d8, 0xc00839cea0}, 0x0, 0x0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:60 +0x5f
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon({0x5d3e7d8, 0xc00839cea0}, 0x3d)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:41 +0x4a
        github.com/cockroachdb/cockroach/pkg/sql/logictest.(*logicTest).processSubtest(0xc006cbfc00, {{0x0, 0x0}, 0xc02cc07e90, 0xc00380dbc0}, {0xc00de4ab80, 0x0})
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:2432 +0x2e2e
        github.com/cockroachdb/cockroach/pkg/sql/logictest.(*logicTest).processTestFile(0xc006cbfc00, {0xc00de4ab80, 0x15}, {{0x49cfc44, 0xc}, 0x3, 0x0, {0x0, 0x0}, {0x0, ...}, ...})
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:1979 +0x5b6
        github.com/cockroachdb/cockroach/pkg/sql/logictest.(*logicTest).runFile(0xc006cbfc00, {0xc00de4ab80, 0x77da8c0}, {{0x49cfc44, 0xc}, 0x3, 0x0, {0x0, 0x0}, {0x0, ...}, ...})
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:3358 +0xda
        github.com/cockroachdb/cockroach/pkg/sql/logictest.RunLogicTestWithDefaultConfig.func1.1(0xc00839cea0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:3566 +0x3d0
        testing.tRunner(0xc00839cea0, 0xc00c9615c0)
        	/usr/local/go/src/testing/testing.go:1259 +0x102
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:1306 +0x35a
--- done: ../../sql/logictest/testdata/logic_test/distsql_automatic_stats with config 3node-tenant: 10 tests, 0 failures
        --- FAIL: TestTenantLogic/3node-tenant/distsql_automatic_stats (46.94s)
=== RUN   TestTenantLogic/3node-tenant
    --- FAIL: TestTenantLogic/3node-tenant (983.84s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantLogic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/logictestccl.TestTenantLogic [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4050510&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4050510&tab=artifacts#/) on master @ [d8278a096c4164b6be0eb5d6a77b2838c527dc84](https://github.com/cockroachdb/cockroach/commits/d8278a096c4164b6be0eb5d6a77b2838c527dc84):


```
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTenantLogic328605770
    test_log_scope.go:148: -- test log scope end --
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestTenantLogic
    logic.go:3584: -- test log scope end --
--- FAIL: TestTenantLogic (1422.70s)
=== RUN   TestTenantLogic/3node-tenant/distsql_automatic_stats
    logic.go:2423: condition failed to evaluate within 45s: 
        ../../sql/logictest/testdata/logic_test/distsql_automatic_stats:66: SELECT DISTINCT ON (column_names) statistics_name, column_names, row_count, distinct_count, null_count
        FROM [SHOW STATISTICS FOR TABLE data] ORDER BY column_names ASC, created DESC
        expected:
            statistics_name  column_names  row_count  distinct_count  null_count
            __auto__         {a,b,c}       1000       1000            0
            __auto__         {a,b}         1000       100             0
            __auto__         {a}           1000       10              0
            __auto__         {b}           1000       10              0
            __auto__         {c}           1000       10              0
            __auto__         {d}           1000       2               730
            
        but found (query options: "colnames,rowsort,retry" -> ignore the following ordering of rows) :
            statistics_name  column_names  row_count  distinct_count  null_count
            __auto__         {a}           1000       10              0
            __auto__         {a,b}         1000       100             0
            __auto__         {a,b,c}       1000       1000            0
            __auto__         {b}           1000       10              0
            __auto__         {c}           1000       10              0
            __auto__         {d}           1000       1               1000
        goroutine 8255506 [running]:
        runtime/debug.Stack()
        	/usr/local/go/src/runtime/debug/stack.go:24 +0x65
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithin({0x5e30038, 0xc02342b040}, 0x0, 0x0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:60 +0x5f
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon({0x5e30038, 0xc02342b040}, 0x3d)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:41 +0x4a
        github.com/cockroachdb/cockroach/pkg/sql/logictest.(*logicTest).processSubtest(0xc02582e400, {{0x0, 0x0}, 0xc00d7da720, 0xc01f547af0}, {0xc009626a40, 0x0})
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:2423 +0x2e2e
        github.com/cockroachdb/cockroach/pkg/sql/logictest.(*logicTest).processTestFile(0xc02582e400, {0xc009626a40, 0x15}, {{0x4aace41, 0xc}, 0x3, 0x0, {0x0, 0x0}, {0x0, ...}, ...})
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:1970 +0x5b6
        github.com/cockroachdb/cockroach/pkg/sql/logictest.(*logicTest).runFile(0xc02582e400, {0xc009626a40, 0x792a1b0}, {{0x4aace41, 0xc}, 0x3, 0x0, {0x0, 0x0}, {0x0, ...}, ...})
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:3349 +0xda
        github.com/cockroachdb/cockroach/pkg/sql/logictest.RunLogicTestWithDefaultConfig.func1.1(0xc02342b040)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:3557 +0x3d0
        testing.tRunner(0xc02342b040, 0xc032dbecc0)
        	/usr/local/go/src/testing/testing.go:1259 +0x102
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:1306 +0x35a
--- done: ../../sql/logictest/testdata/logic_test/distsql_automatic_stats with config 3node-tenant: 10 tests, 0 failures
        --- FAIL: TestTenantLogic/3node-tenant/distsql_automatic_stats (48.88s)
=== RUN   TestTenantLogic/3node-tenant
    --- FAIL: TestTenantLogic/3node-tenant (1416.14s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantLogic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/logictestccl.TestTenantLogic [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4149826&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4149826&tab=artifacts#/) on master @ [912964e02ddd951c77d4f71981ae18b3894e9084](https://github.com/cockroachdb/cockroach/commits/912964e02ddd951c77d4f71981ae18b3894e9084):

Fatal error:

```
panic: streamer-2-unlimited-4: no bytes in account to release, current 1024, free 1025 [recovered]
	panic: streamer-2-unlimited-4: no bytes in account to release, current 1024, free 1025

```
Stack: 

```
goroutine 141601611 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Recover(0xc01487ccf0, {0x5fcc2d8, 0xc01c3441a0})
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:251 +0x94
panic({0x48b38e0, 0xc019752528})
	/usr/local/go/src/runtime/panic.go:1038 +0x215
github.com/cockroachdb/cockroach/pkg/util/log/logcrash.ReportOrPanic({0x5fcc2d8, 0xc01c3441a0}, 0xc01954a000, {0x4d39956, 0x5}, {0xc028199a08, 0x45bab00, 0xc02c693bc0})
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/logcrash/crash_reporting.go:376 +0x1af
github.com/cockroachdb/cockroach/pkg/util/mon.(*BoundAccount).Shrink(0xc0236e3f38, {0x5fcc2d8, 0xc01c3441a0}, 0x401)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/mon/bytes_usage.go:636 +0x127
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvstreamer.(*budget).release(0xc02c693bc0, {0x5fcc2d8, 0xc01c3441a0}, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvstreamer/budget.go:117 +0xb0
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvstreamer.(*workerCoordinator).performRequestAsync.func1({0x5fcc2d8, 0xc01c3441a0})
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvstreamer/streamer.go:1025 +0x3af
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2()
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:488 +0x12a
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:478 +0x445
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestTenantLogic/3node-tenant/cast/assignment_casts_insert_do_update
*
* ERROR: a panic has occurred!
* streamer-2-unlimited-4: no bytes in account to release, current 1024, free 1025
* (1) attached stack trace
*   -- stack trace:
*   | runtime.gopanic
*   | 	/usr/local/go/src/runtime/panic.go:1038
*   | [...repeated from below...]
* Wraps: (2) attached stack trace
*   -- stack trace:
*   | github.com/cockroachdb/cockroach/pkg/util/log/logcrash.ReportOrPanic
*   | 	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/logcrash/crash_reporting.go:374
*   | github.com/cockroachdb/cockroach/pkg/util/mon.(*BoundAccount).Shrink
*   | 	/go/src/github.com/cockroachdb/cockroach/pkg/util/mon/bytes_usage.go:636
*   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvstreamer.(*budget).release
*   | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvstreamer/budget.go:117
*   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvstreamer.(*workerCoordinator).performRequestAsync.func1
*   | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvstreamer/streamer.go:1025
*   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2
*   | 	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:488
*   | runtime.goexit
*   | 	/usr/local/go/src/runtime/asm_amd64.s:1581
* Wraps: (3) streamer-2-unlimited-4: no bytes in account to release, current 1024, free 1025
* Error types: (1) *withstack.withStack (2) *withstack.withStack (3) *errutil.leafError
*
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=deadlock

- GOFLAGS=-parallel=4
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantLogic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/logictestccl.TestTenantLogic [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4184818&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4184818&tab=artifacts#/) on master @ [6aa6c727b1d990bc0e3e8fbc36e25fc358ba39c1](https://github.com/cockroachdb/cockroach/commits/6aa6c727b1d990bc0e3e8fbc36e25fc358ba39c1):


```
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTenantLogic3296889369
    test_log_scope.go:148: -- test log scope end --
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestTenantLogic
    logic.go:3613: -- test log scope end --
--- FAIL: TestTenantLogic (1394.68s)
=== RUN   TestTenantLogic/3node-tenant/distsql_automatic_stats
    logic.go:2436: condition failed to evaluate within 45s: 
        ../../sql/logictest/testdata/logic_test/distsql_automatic_stats:66: SELECT DISTINCT ON (column_names) statistics_name, column_names, row_count, distinct_count, null_count
        FROM [SHOW STATISTICS FOR TABLE data] ORDER BY column_names ASC, created DESC
        expected:
            statistics_name  column_names  row_count  distinct_count  null_count
            __auto__         {a,b,c}       1000       1000            0
            __auto__         {a,b}         1000       100             0
            __auto__         {a}           1000       10              0
            __auto__         {b}           1000       10              0
            __auto__         {c}           1000       10              0
            __auto__         {d}           1000       2               730
            
        but found (query options: "colnames,rowsort,retry" -> ignore the following ordering of rows) :
            statistics_name  column_names  row_count  distinct_count  null_count
            __auto__         {a}           1000       10              0
            __auto__         {a,b}         1000       100             0
            __auto__         {a,b,c}       1000       1000            0
            __auto__         {b}           1000       10              0
            __auto__         {c}           1000       10              0
            __auto__         {d}           1000       1               1000
        goroutine 2070700 [running]:
        runtime/debug.Stack()
        	/usr/local/go/src/runtime/debug/stack.go:24 +0x65
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithin({0x601a6d8, 0xc010a54000}, 0x0, 0x0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:60 +0x5f
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon({0x601a6d8, 0xc010a54000}, 0x3d)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:41 +0x4a
        github.com/cockroachdb/cockroach/pkg/sql/logictest.(*logicTest).processSubtest(0xc01963c000, {{0x0, 0x0}, 0xc01223c4e0, 0xc010465540}, {0xc00c66ea80, 0x0})
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:2436 +0x2e2e
        github.com/cockroachdb/cockroach/pkg/sql/logictest.(*logicTest).processTestFile(0xc01963c000, {0xc00c66ea80, 0x15}, {{0x4c4b632, 0xc}, 0x3, 0x0, {0x0, 0x0}, {0x0, ...}, ...})
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:1983 +0x5b6
        github.com/cockroachdb/cockroach/pkg/sql/logictest.(*logicTest).runFile(0xc01963c000, {0xc00c66ea80, 0x7b810f0}, {{0x4c4b632, 0xc}, 0x3, 0x0, {0x0, 0x0}, {0x0, ...}, ...})
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:3378 +0xda
        github.com/cockroachdb/cockroach/pkg/sql/logictest.RunLogicTestWithDefaultConfig.func1.1(0xc010a54000)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:3586 +0x3d0
        testing.tRunner(0xc010a54000, 0xc01a80e060)
        	/usr/local/go/src/testing/testing.go:1259 +0x102
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:1306 +0x35a
--- done: ../../sql/logictest/testdata/logic_test/distsql_automatic_stats with config 3node-tenant: 10 tests, 0 failures
        --- FAIL: TestTenantLogic/3node-tenant/distsql_automatic_stats (48.35s)
=== RUN   TestTenantLogic/3node-tenant
    --- FAIL: TestTenantLogic/3node-tenant (1382.62s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantLogic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/logictest/testdata/logic_test/distsql_automatic_stats
+++ b/pkg/sql/logictest/testdata/logic_test/distsql_automatic_stats
@@ -1,4 +1,8 @@
-# LogicTest: !metamorphic
+# LogicTest: !metamorphic !3node-tenant
+
+# Note: this test is disabled on 3node-tenant because it sometimes causes one of
+# the UPDATE statements below (where we update more than 20% of the table) to be
+# flaky. See comments there for details.
 
 # Disable automatic stats
 statement ok
@@ -63,6 +67,18 @@ SET CLUSTER SETTING sql.stats.automatic_collection.enabled = true
 statement ok
 UPDATE data SET d = 12 WHERE d = 10
 
+# Note: this statement is flaky on 3node-tenant (which is why that config is
+# disabled). Normally, the entire UPDATE in the previous statement is performed
+# as a single batch, since the total number of rows updated is 270, which is
+# less than the kv batch size of 100k rows (the kv batch size doesn't change in
+# this test since the metamorphic config is disabled). The target row count to
+# trigger a stats refresh is 205, since 20% of 1000 rows is 200, and
+# sql.stats.automatic_collection.min_stale_rows is set to 5 for logictests.
+# Since 270 > 205, we should always trigger a refresh.
+#
+# For some reason, 3node-tenant occasionally splits the UPDATE into 4 pieces,
+# with each one affecting at most 88 rows. Since 88 < 205, the refresh is not
+# guaranteed, making this test flaky.
 query TTIII colnames,rowsort,retry
 SELECT DISTINCT ON (column_names) statistics_name, column_names, row_count, distinct_count, null_count
 FROM [SHOW STATISTICS FOR TABLE data] ORDER BY column_names ASC, created DESC

